action/protocol/staking/candidate_statemanager.go

```diff
@@ -94,6 +94,56 @@ func (csm *candSM) Upsert(d *Candidate) error {
 	return nil
 }
 
+func (csm *candSM) Commit() error {
+	if err := csm.CandidateCenter.Commit(); err != nil {
+		return err
+	}
+
+	// write update view back to state factory
+	return csm.WriteView(protocolID, csm.CandidateCenter)
+}
+
+func getOrCreateCandCenter(sr protocol.StateReader) (CandidateCenter, error) {
+	c, err := getCandCenter(sr)
+	if err != nil {
+		if errors.Cause(err) == protocol.ErrNoName {
+			// the view does not exist yet, create it
+			cc, err := createCandCenter(sr)
+			return cc, err
+		}
+		return nil, err
+	}
+	return c, nil
+}
+
+func getCandCenter(sr protocol.StateReader) (CandidateCenter, error) {
+	v, err := sr.ReadView(protocolID)
+	if err != nil {
+		return nil, err
+	}
+
+	if center, ok := v.(CandidateCenter); ok {
+		return center, nil
+	}
+	return nil, errors.Wrap(ErrTypeAssertion, "expecting CandidateCenter")
+}
+
+func createCandCenter(sr protocol.StateReader) (CandidateCenter, error) {
+	all, err := loadCandidatesFromSR(sr)
+	if err != nil {
+		return nil, err
+	}
+
+	center := NewCandidateCenter()
+	if err := center.SetDelta(all); err != nil {
+		return nil, err
+	}
+	if err := center.Commit(); err != nil {
+		return nil, err
+	}
```

Comment:
why commit here?

Reply:
center := NewCandidateCenter() at line 137 is an empty struct

say there are 5 candidates at beginning, they are read out from sf/DB, and add to it as the Delta, Commit() writes the 5 candidates into the center, then WriteView save it to sf as the starting point

Reply:
I got what you want to do
`return NewCandidateCenter(all)`